### DISCUSSION TITLE
.clang-format

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2584] [General]
Has it ever been considered to create .clang-format file in the project root?

**Comment by sreimers** [comment]
I  have one [here](https://github.com/Studio-Link/next/blob/main/.clang-format), but it only can take care of 80-90% of the format rules we have and if someone use auto format, we get pull requests with many unrelated changes.

**Reply by juha-h** [reply]
Yes, like clang mode in emacs checks the format when the file is saved and if something does not match, it corrects the format automatically.
